src/components/swap/SwapDetailsDropdown.tsx

```diff
@@ -102,13 +103,23 @@ export default function SwapDetailsDropdown({ trade, syncing, loading, allowedSl
   const theme = useTheme()
   const { chainId } = useWeb3React()
   const [showDetails, setShowDetails] = useState(false)
+  const trace = useTrace()
+
+  const analyticsProps = useMemo(
+    () => ({
+      ...(trade ? formatEventPropertiesForTrade(trade, allowedSlippage) : {}),
+      ...trace,
+    }),
+    [allowedSlippage, trace, trade]
+  )
```

Comment:
dont need to memoize this - the events array is already a new object on every render so this doesnt add anything perf wise